src/operator/contrib/count_sketch.cu

```diff
@@ -165,7 +165,8 @@ inline void CountSketchBackward(const Tensor<gpu, 2, DType> &in_grad,
                             nthreads, in_grad_ptr+bstart*in_dim, h_ptr,
                             s_ptr, out_grad_ptr+bstart*out_dim, batchlen,
                             in_dim, out_dim);
-    MSHADOW_CUDA_POST_KERNEL_CHECK(sketch_backward_kernel);
+    cudaError_t err = cudaDeviceSynchronize();
```

Comment:
DRY - this check repeats in line 132 (and maybe elsewhere?) - did you consider refactoring into a common function?

Reply:
This is the only place where we want to use cudaDeviceSynchronize, if I define an extra function it's also 4 extra lines in total, and does not make any difference.

Reply:
Oh it's actually 7 extra lines where I need 5 lines to define the function and 2 extra calls to the function. Now it's only 4 extra lines in total.

Reply:
DRY is not only about lines of code. Consider a case where there is an issue with the flow in line 168, and the developer decides to change it from cudaDeviceSynchronize to another API. Without DRY, the developer will need to remember to look for other places in the code that call the API. Sometimes, this will be missed. With DRY, different parts of the code are funneled to the one function or whatever that has this logic, and by updating in one location all code paths are updated. Makes sense?